x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/TDigestFieldMapper.java

```diff
@@ -340,15 +343,22 @@ public void parse(DocumentParserContext context) throws IOException {
             TDigestParser.ParsedHistogram parsedHistogram = TDigestParser.parse(fullPath(), subParser);
 
             BytesStreamOutput streamOutput = new BytesStreamOutput();
+
+            streamOutput.writeDouble(parsedHistogram.min());
+            streamOutput.writeDouble(parsedHistogram.max());
+            streamOutput.writeDouble(parsedHistogram.sum());
+            streamOutput.writeLong(parsedHistogram.count());
```

Comment:
Storing these in a single `BinaryDocValues` prevents us from optimizing queries on loading only the required data. E.g. a `MIN(histogram)` will have inferior performance to `MIN(some_double_column)`because it (a) loads more data and (b) needs to extract the values from the bytes instead of being able to operate directly on the memory as is.

In addition this will use more storage, because it can't use our optimized numeric doc values encodings.
(At least until we have some compression on `BinaryDocValue`)

I guess this is a known tradeoff for now and something we accept to move faster? Nothing will prevent us from optimizing this later.